core/txpool/legacypool/legacypool.go

```diff
@@ -1670,6 +1678,57 @@ func (pool *LegacyPool) demoteUnexecutables() {
 	}
 }
 
+// GetInclusionList returns an inclusion list from the pool containing pairs
+// of transaction summary and data which are executable.
+func (pool *LegacyPool) GetInclusionList() *types.InclusionList {
+	// The following conditions should be satisfied while filling the inclusion list.
+	// 		1. The size of IL should not exceed `maxTxsPerInclusionList`
+	//
+	// 		2. The total gas limit should not exceed `maxGasPerInclusionList`
+	//
+	// 		3. The `maxFeePerGas` of all the transactions should be 1.125 times
+	//			the `baseFee` of the current block to account for increase in the
+	// 			next block due to EIP 1559.
+	summaries := make([]*types.InclusionListEntry, 0, maxTxsPerInclusionList)
+	transactions := make([]*types.Transaction, 0, maxTxsPerInclusionList)
+
+	// TODO(manav2401): Not sure what's the best way to fetch transactions for inclusion list.
+	// Few possibilities are: prioritise locals first, borrow the tx ordering
+	// logic from miner if we want to choose best transactions.
```

Comment:
Marius had an idea here with a tx drop cache that tracks when transactions are reorged out of blocks: https://github.com/ethereum/go-ethereum/pull/26829/files

I think this is a pretty particular use case though. You could probably create a list of transactions ordered by i) age and ii) how long they've been able to pay the base fee. ii) is a bit more difficult to ascertain.